backend/routers/buildings.js

```diff
@@ -1,97 +1,100 @@
 const { Router } = require('express');
+const createError = require('http-errors');
+const { BuildingModel, ResourceModel } = require('../models');
+const { auth } = require('../middlewares');
 
 const router = Router();
 
 const {
   townhallRule, farmRule, mineRule, academyRule, foxRule,
 } = require('../rules');
 
-const myBuildings = {
-  buildings: [
-    {
-      id: 1,
-      type: 'Townhall',
-      level: 1,
-      hp: 1,
-      started_at: 12345789,
-      finished_at: 12399999,
-    }, {
-      id: 2,
-      type: 'Academy',
-      level: 1,
-      hp: 1,
-      started_at: 12345789,
-      finished_at: 12399999,
-    }, {
-      id: 3,
-      type: 'Farm',
-      level: 1,
-      hp: 1,
-      started_at: 12345789,
-      finished_at: 12399999,
-    }, {
-      id: 4,
-      type: 'Mine',
-      level: 1,
-      hp: 1,
-      started_at: 12345789,
-      finished_at: 12399999,
-    },
-  ],
+const typeToRules = {
+  Townhall: townhallRule,
+  Academy: academyRule,
+  Farm: farmRule,
+  Mine: mineRule,
 };
 
-function getBuildings(req, res) {
-  return res.status(200).send(myBuildings);
+async function getBuildings(req, res, next) {
+  const { _id: owner } = req.user;
+  try {
+    const buildings = await BuildingModel.find({ owner });
+    res.send({ buildings });
+  } catch (error) {
+    next(error);
+  }
 }
 
-function getBuildingById(req, res) {
-  const buildingId = Number.parseInt(req.params.buildingId, 10);
-  if (Number.isNaN(buildingId) || buildingId <= 0) {
-    return res.sendStatus(400);
+async function createBuilding(req, res, next) {
+  const { _id: owner } = req.user;
+  const { buildingType } = req.params;
+  try {
+    if (!buildingType) {
+      throw createError(400, 'Type is required');
+    }
+    if (buildingType !== 'Mine' && buildingType !== 'Farm') {
+      throw createError(400, 'Wrong type');
+    }
+
+    const moneyRequired = parseInt(typeToRules[buildingType].constructionCost, 10);
+    const resourceOfUser = await ResourceModel.find({ owner });
+    const [moneyInHand] = resourceOfUser
+      .filter(({ type }) => (type === 'gold'))
+      .map(({ amount }) => amount);
+    if (moneyRequired > moneyInHand) {
+      throw createError(400, 'You don\'t have enough money');
+    }
+
+    const minusGold = await ResourceModel.findOneAndUpdate(
+      { owner, type: 'gold' },
+      { $inc: { amount: -{ moneyRequired } } },
+      { new: true, fields: '-_id' },
+    ).exec();
+    const createNewBuilding = await BuildingModel.create({
+      type: buildingType,
+      owner,
+    });
+    res.send({ minusGold, createNewBuilding });
```

Comment:
```suggestion
    res.send({ minusGold, newBuilding });
```

if the name contains 'create' in it, it suggests it's a function, but we want to send back a building now, and not a function